helper/schema/resource.go

```diff
@@ -330,6 +330,10 @@ func (r *Resource) Apply(
 		return s, append(diags, diag.FromErr(err))
 	}
 
+	if s != nil && data != nil {
```

Comment:
Assuming its okay for these to be nil in certain situations? No need for logging warnings or panicking?

Reply:
I suppose in the context of provider schema being an entirely optional feature sanity checking nil like this is safe and we don't really need to think about why something might be nil, the rest of the SDK should concern itself with that, not this feature.

Reply:
In most cases, this will be nil. This will only be non-nil when a provider_meta block is set for this provider.